src/WebJobs.Extensions.DurableTask/OutOfProcMiddleware.cs

```diff
@@ -0,0 +1,363 @@
+                        context.ContinuedAsNew,
+                        FunctionType.Orchestrator,
+                        isReplay: false);
+#if !FUNCTIONS_V1
+                    DurableTaskExtension.TagActivityWithOrchestrationStatus(
+                        OrchestrationRuntimeStatus.Completed,
+                        instance.InstanceId);
+#endif
+                    await this.LifeCycleNotificationHelper.OrchestratorCompletedAsync(
+                        this.Options.HubName,
+                        functionName.Name,
+                        instance.InstanceId,
+                        context.ContinuedAsNew,
+                        isReplay: false);
+                }
+                else
+                {
+                    this.TraceHelper.FunctionAwaited(
+                        this.Options.HubName,
+                        functionName.Name,
+                        FunctionType.Orchestrator,
+                        instance.InstanceId,
+                        isReplay: false);
+                }
+            }
+            else
+            {
+                string exceptionDetails = functionResult.Exception.ToString();
+
+                this.TraceHelper.FunctionFailed(
+                    this.Options.HubName,
+                    functionName.Name,
+                    instance.InstanceId,
+                    exceptionDetails,
+                    FunctionType.Orchestrator,
+                    isReplay: false);
+
+                await this.LifeCycleNotificationHelper.OrchestratorFailedAsync(
+                    this.Options.HubName,
+                    functionName.Name,
+                    instance.InstanceId,
+                    exceptionDetails,
+                    isReplay: false);
+
+                orchestratorResult = OrchestratorExecutionResult.ForFailure(
+                    message: $"Function '{functionName}' failed with an unhandled exception.",
+                    functionResult.Exception);
+            }
+
+            // Send the result of the orchestrator function to the DTFx dispatch pipeline.
+            // This allows us to bypass the default, in-process execution and process the given results immediately.
+            dispatchContext.SetProperty(orchestratorResult);
+        }
+
+        /// <summary>
+        /// Durable Task Framework activity middleware that invokes an out-of-process orchestrator function.
+        /// </summary>
+        /// <param name="dispatchContext">This middleware context provided by the framework that contains information about the activity.</param>
+        /// <param name="next">The next middleware handler in the pipeline.</param>
+        /// <exception cref="SessionAbortedException">Thrown if there is a recoverable error in the Functions runtime that's expected to be handled gracefully.</exception>
+        public async Task CallActivityAsync(DispatchMiddlewareContext dispatchContext, Func<Task> next)
```

Comment:
Are we going to need this middleware for every API? If so, am I right in understanding that currently only `CallActivityAsync` is supported in dotnet-isolated`?

Reply:
I'm not sure what you mean by _every API_, but yes, we need this middleware to run for _every activity invocation_. I'm also not sure I understand your question about dotnet-isolated or how it relates to this middleware.